jwst/extract_2d/nirspec.py

```diff
@@ -82,6 +90,16 @@ def nrs_extract2d(input_model, slit_name=None):
             # The overall subarray offset is recorded in model.meta.subarray.
             set_slit_attributes(new_model, slit, xlo, xhi, ylo, yhi)
 
+            if (new_model.meta.exposure.type.lower() == 'nrs_fixedslit') and \
+                (slit.name == input_model.meta.instrument.fixed_slit):
+                try:
+                    get_source_xpos(new_model)
+                except DitherMetadataError as e:
+                    log.warning(str(e))
+                    log.warning("Setting source position in slit to 0.0, 0.0")
+                    new_model.source_ypos = 0.0
+                    new_model.source_xpos = 0.0
+
```

Comment:
Can we add an `else` clause here to set the position to 0.0 for non-primary slits?  That would help ensure we never get garbage positions in later steps.